app/(main)/page.tsx

```diff
@@ -1,15 +1,67 @@
 'use client';
 
 import { useQuery } from '@tanstack/react-query';
+import { useState } from 'react';
+
 import { groupMembersQueryKey } from '@/queries/groupMembersQueryKey';
-import { getGroupMembers, GroupMember } from '@/apis/groupMembers';
+import { getGroupMembers, GroupMember, Position } from '@/apis/groupMembers';
+
+import CommonInput from '@/components/common/CommonInput';
 
 export default function Home() {
-  const { data, error, isLoading } = useQuery<GroupMember[]>({
+  const { data } = useQuery<GroupMember[]>({
     queryKey: groupMembersQueryKey.getGroupMembers(1),
     queryFn: () => getGroupMembers(1),
     staleTime: 1000 * 60 * 60,
   });
 
-  return <div></div>;
+  const [searchUser, setSearchUser] = useState('');
+  const searchData = data?.filter((groupMember) => {
```

Comment:
```suggestion
  const searchData = useMemo(() => {
    if (!data) return [];
    return data.filter((groupMember) => {
      if (!searchUser) return true;
      return (
        groupMember.name.includes(searchUser.trim()) ||
        groupMember.nickname.includes(searchUser.trim())
      );
    });
  }, [data, searchUser]);
```

1. useMemo를 사용하면 좀 더 성능 측면에서 좋지않을까요??
2. data?.filter 보다 명시적으로 if문으로 검사하면 어떨까요?? (이건 코드 취향같기도 하네요 ㅎㅎ)